includes/class-sensei-learner.php

```diff
@@ -111,6 +114,40 @@ public function delete_all_user_activity( $user_id = 0 ) {
 		return $dataset_changes;
 	}
 
+	/**
+	 * Remove duplicate progress comments to mitigate duplicate enrollment issue.
+	 *
+	 * Hooked into sensei_log_activity_after
+	 *
+	 * @since 3.7.0
+	 * @access private
+	 *
+	 * @param array $args
+	 */
+	public function remove_duplicate_progress( $args ) {
+		add_action(
+			'shutdown',
+			function() use ( $args ) {
+				global $wpdb;
+
+				// This query remove all comments, but the first one, which match the context conditions.
+				// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
+				$wpdb->query(
+					$wpdb->prepare(
+						"DELETE c1.*
+							FROM $wpdb->comments as c1
```

Comment:
This could possibly leave abandoned meta in `{$prefix}comment_meta` and stale cache/comment counts. Although a different beast, in my [tools PR](https://github.com/Automattic/sensei/pull/3842/files#diff-c538767d485d10cc90289be4e5c8a60e0addfa7e2d330099f7463917bd3c3985R58) I separated the queries so I could manually flush the cache and delete the meta.

Reply:
Hey Jake! Great catch! I just didn't call the `Sensei()->flush_comment_counts_cache` because it's called immediately after the action. ;)

I took the opportunity to also send an event (`remove_duplicate_progress_comments`), so we can monitor if the issue is still happening, of if it's fixed by [this PR](https://github.com/Automattic/sensei/pull/3838). cc @donnapep